addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.addons.resource.models.utils import string_to_datetime
+DAYS = ['monday_location_id', 'tuesday_location_id', 'wednesday_location_id', 'thursday_location_id', 'friday_location_id', 'saturday_location_id', 'sunday_location_id']
+
+class HrHomeworking(models.Model):
+    _name = "hr.homeworking"
+
+    def _default_employee(self):
+        if 'default_employee' in self.env.context:
+            return self.env.context.default_employee
+        return self.env['hr.employee'].search([('user_id.id','=', self.env.user.id)])
```

Comment:
Also I don't see where 'default_employee' is coming from, where it is put in the context, but if instead you could put `default_employee_id` and it will automatically set a field. So, no need to manually set ^^